### PR TITLE
Stop passing components as properties in the Project view

### PR DESCRIPTION
**Pull request by ciyer**
The is preparation for further work on #2415. 

/deploy #persist #cypress renku=renku-ui-3.4.1-tests

### CODE REVIEW

File: client/src/project/datasets/ProjectDatasets.test.js [hunk 1]
```diff
@@ -72,44 +62,6 @@ describe("rendering", () => {
     spy.mockRestore();
   });
 
-  it("renders datasets list without crashing", async () => {
```
Comment (lorenzo-cavazzi) [inline client/src/project/datasets/ProjectDatasets.test.js:75]: Should we replace this with an e2e test?
Or would you like to do it later when updating the dataset components? I assume the other sub-components should have e2e tests as well

**Review by lorenzo-cavazzi — APPROVED** [review]
This looks very good! I added an inline comment; feel free to address it either here or in the follow-up datasets refactoring issue.

File: client/src/project/datasets/ProjectDatasets.test.js (hunk 1 above)
Comment (ciyer) [reply client/src/project/datasets/ProjectDatasets.test.js:75]: > Should we replace this with an e2e test? Or would you like to do it later when updating the dataset components? I assume the other sub-components should have e2e tests as well

Yes, this is already covered by the `datasets.spec.ts` test case.

**Review by lorenzo-cavazzi — DISMISSED** [review]
Lgtm! :rocket:

**Review by lorenzo-cavazzi — APPROVED** [review]

**Pull request merged**